bodhi/server/consumers/greenwave.py

```diff
@@ -68,5 +68,28 @@ def __call__(self, message: fedora_messaging.api.Message):
                 log.debug(f"Couldn't find build {subject_identifier} in DB")
                 return
 
-            log.info(f"Updating the test_gating_status for: {build.update.alias}")
-            build.update.update_test_gating_status()
+            update = build.update
+            log.info(f"Updating the test_gating_status for: {update.alias}")
+            if len(update.builds) > 1:
+                update.update_test_gating_status()
+            else:
+                update.test_gating_status = self._extract_gating_status(msg)
+
+    def _extract_gating_status(self, msg):
+        """
+        Extract gating information from the Greenwave message and return it.
+
+        Returns:
+            TestGatingStatus:
+                - TestGatingStatus.ignored if no tests are required
+                - TestGatingStatus.failed if policies are not satisfied
+                - TestGatingStatus.passed if policies are satisfied, and there
+                  are required tests
+        """
+        if not msg['policies_satisfied']:
```

Comment:
just to be on the safe side: ``msg.get("policies_satisfied")`` here?

Reply:
or better, we check if that key is in the dict first and bail if it isn't then move on to your code as is :)

Reply:
Sure, why not, I added that.